cpp/src/utilities/trie.cuh

```diff
@@ -33,6 +33,7 @@ struct SerialTrieNode {
 	int16_t children_offset = -1;
 	char character;
 	bool is_leaf;
+    SerialTrieNode() = default; // FIXME This is necessary for a Thrust bug on CentOS7 + CUDA10
```

Comment:
AFAIK calling this constructor will create an object with undefined values. Given how easy it is to overlook the calls to the default constructor, I would like it to behave consistently.

Please initialize *character* and *is_leaf*, either in an init list, or inline. Using *trie_terminating_character* would make sense for default value of *character*, *is_leaf* can be false.

Reply:
Sure, my hope is that the default can just be deleted once the Thrust bug is resolved, but makes sense to make it safe in the meantime.

I'll add default in-class initializers.